manual/developer/request_execution/README.md

```diff
@@ -0,0 +1,314 @@
+                 | 1 per node +-------------+
+                 +------------+ ChannelPool |
+                 |            +----+--------+
+                 |                 |
+                 |                 |  n +---------------+
+                 |                 +----+ DriverChannel |
+                 |                      +---------------+
+                 |
+                 |          1 +--------------------------+
+                 +------------+ RequestProcessorRegistry |
+                              +----+---------------------+
+                                   |
+                                   |  n +---------------------------+
+                                   +----+ RequestProcessor          |
+                                        +---------------------------+
+                                        | ResultT process(RequestT) |
+                                        +---------------------------+
+```
+
+`DefaultSession` contains the session implementation. It follows the [confined inner
+class](../common/concurrency/#cold-path) pattern to simplify concurrency.
+
+### Connection pooling
+
+```ditaa
++----------------------+       1 +------------+
+| ChannelPool          +---------+ ChannelSet |
++----------------------+         +-----+------+
+| DriverChannel next() |               |
++----------+-----------+              n|
+           |                    +------+--------+
+          1|                    | DriverChannel |
+    +------+-------+            +---------------+
+    | Reconnection |
+    +--------------+
+```
+
+`ChannelPool` handles the connections to a given node, for a given session. It follows the [confined
+inner class](../common/concurrency/#cold-path) pattern to simplify concurrency. There are a few
+differences compared to the 3.x implementation:
+
+#### Fixed size
+
+The pool has a fixed number of connections, it doesn't grow or shrink dynamically based on current
+usage. In other words, there is no more "max" size, only a "core" size.
+
+However, this size is specified in the configuration. If the value is changed at runtime, the driver
+will detect it, and trigger a resize of all active pools.
+
+The rationale for removing the dynamic behavior is that it introduced a ton of complexity in the
+implementation and configuration, for unclear benefits: if the load fluctuates very rapidly, then
+you need to provision for the max size anyway, so you might as well run with all the connections all
+the time. If on the other hand the fluctuations are rare and predictable (e.g. peak for holiday
+sales), then a manual configuration change is good enough.
+
+#### Wait-free
+
+To get a connection to a node, client code calls `ChannelPool.next()`. This returns the less busy
+connection, based on the the `getAvailableIds()` counter exposed by
+[InFlightHandler](netty_pipeline/#in-flight-handler).
+
+If all connections are busy, there is no queuing; the driver moves to the next node immediately. The
```

Comment:
How does this articulate with load balancing policies? IOW, am I right to infer that if the LBP picks nodeA, we might end up querying nodeB if all connections to nodeA are busy?